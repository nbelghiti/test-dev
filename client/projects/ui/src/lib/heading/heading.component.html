<div
	class="c-heading {{ classes }}"
	[ngClass]="{'c-heading--inline': inline}"
>
	<ng-container [ngSwitch]="level">
		<h1 *ngSwitchCase="HeadingLevel.heading1">{{title}}</h1>
		<h2 *ngSwitchCase="HeadingLevel.heading2">{{title}}</h2>
		<h3 *ngSwitchCase="HeadingLevel.heading3">{{title}}</h3>
	</ng-container>

	<div class="c-heading__meta">
		<i
			*ngIf="loading; else withContent"
			class="fas fa-{{IconName.spinner}} fa-spin c-heading__loader"
		>
		</i>

		<ng-template #withContent>
			<ng-content></ng-content>
		</ng-template>
	</div>
</div>
