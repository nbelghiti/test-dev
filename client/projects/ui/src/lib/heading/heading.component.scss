// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Heading
// =============================================================================
.c-heading {
	display: inline-flex;
	flex-direction: column;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-heading__loader {
	color: $primary-color;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-heading--inline {
	flex-direction: row;
	align-items: baseline;

	.c-heading__meta {
		margin-top: 0;
		margin-left: 1rem;
	}
}
