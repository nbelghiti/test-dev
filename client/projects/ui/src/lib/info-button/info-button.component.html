<div class="c-info-button">
	<span class="c-info-button__icon">
		<ui-icon [name]="icon" size="large"></ui-icon>
	</span>

	<ui-heading
		class="c-info-button__title"
		level="3"
		[title]="title"
	></ui-heading>
</div>
