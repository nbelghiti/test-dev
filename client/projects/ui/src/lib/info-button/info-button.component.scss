// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Info Link
// =============================================================================
.c-info-button {
	display: flex;
	align-items: center;
	width: 100%;
	padding: 1.5rem 3rem;

	text-decoration: none;
	text-align: left;

	border-bottom: $border-width-base solid $border-color-grey-76;

	background-color: $background-color-grey-96;

	cursor: pointer;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-info-button__icon {
	display: flex;
	justify-content: center;
	width: 2.4rem;
	margin-right: 1.6rem;

	line-height: $line-height-normalize;

	transition: color $animation-base;
}

.c-info-link__title {
	transition: color $animation-base;
}
