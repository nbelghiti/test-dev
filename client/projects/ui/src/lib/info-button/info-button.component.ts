import { Component, Input } from '@angular/core';

import { IconName } from '../icon/icon.types';

@Component({
	selector: 'ui-info-button',
	templateUrl: './info-button.component.html',
	styleUrls: ['./info-button.component.scss']
})

export class InfoButtonComponent {
	@Input() public icon: IconName;
	@Input() public title: string;
}
