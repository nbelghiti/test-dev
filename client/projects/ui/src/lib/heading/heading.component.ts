import { Component, Input } from '@angular/core';

import { IconName } from '../icon/icon.types';

import { HeadingLevel } from './heading.types';

@Component({
	selector: 'ui-heading',
	templateUrl: './heading.component.html',
	styleUrls: ['./heading.component.scss']
})

export class HeadingComponent {
	@Input() public title: string;
	@Input() public classes: string = '';
	@Input() public inline: boolean = false;
	@Input() public loading: boolean = false;
	@Input() public level: HeadingLevel = HeadingLevel.heading1;

	public HeadingLevel: typeof HeadingLevel = HeadingLevel;
	public IconName: typeof IconName = IconName;
}
