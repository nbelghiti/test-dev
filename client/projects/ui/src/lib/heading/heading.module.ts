import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { HeadingComponent } from './heading.component';

@NgModule({
	imports: [
		CommonModule,
	],
	declarations: [
		HeadingComponent,
	],
	exports: [
		HeadingComponent,
	],
})
export class HeadingModule {}
