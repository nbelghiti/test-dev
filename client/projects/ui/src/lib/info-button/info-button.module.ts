import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { HeadingModule } from '../heading/heading.module';
import { IconModule } from '../icon/icon.module';

import { InfoButtonComponent } from './info-button.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		HeadingModule,
	],
	declarations: [
		InfoButtonComponent,
	],
	exports: [
		InfoButtonComponent,
	],
})
export class InfoButtonModule {}
