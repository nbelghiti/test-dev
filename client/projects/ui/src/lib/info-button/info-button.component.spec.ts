import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { HeadingComponent } from '../heading/heading.component';
import { IconComponent } from '../icon/icon.component';

import { InfoButtonComponent } from './info-button.component';

describe('InfoButtonComponent', () => {
	let component: InfoButtonComponent;
	let fixture: ComponentFixture<InfoButtonComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [
				InfoButtonComponent,
				IconComponent,
				HeadingComponent,
			],
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(InfoButtonComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});

