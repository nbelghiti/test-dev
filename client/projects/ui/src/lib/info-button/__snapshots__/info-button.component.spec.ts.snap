// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoButtonComponent Should render correctly 1`] = `
<ui-info-button>
  <div
    class="c-info-button"
  >
    <span
      class="c-info-button__icon"
    >
      <ui-icon
        class="c-icon"
        ng-reflect-size="large"
        size="large"
      >
        <i
          class="c-icon__glyph c-icon__glyph--large  fas fa-"
        />
      </ui-icon>
    </span>
    <ui-heading
      class="c-info-button__title"
      level="3"
      ng-reflect-level="3"
    >
      <div
        class="c-heading"
        ng-reflect-klass="c-heading "
        ng-reflect-ng-class="[object Object]"
      >
        
        
        
        
        <h3>
          
        </h3>
        <div
          class="c-heading__meta"
        >
          
          
          
        </div>
      </div>
    </ui-heading>
  </div>
</ui-info-button>
`;
